presto-docs/src/main/sphinx/functions/geospatial.rst

```diff
@@ -109,6 +113,26 @@ Operations
 
     Returns the geometry value that represents the point set symmetric difference of two geometries.
 
+.. function:: ST_Buffer(Geometry, double) -> Geometry
+
+    Returns the geometry value that represents all points whose distance from any point of an input geometry is less than or equal to a specified distance
+
+.. function:: point_to_bing_tile(Geometry, int) => varchar
+
+    Given a (longitude, latitude) point and a zoom level, returns a QuadKey for the containing Bing tile. Latitude must be within [-85.05112878, 85.05112878] range. Longitude must be within [-180, 180] range. Zoom levels from 1 to 23 are supported.
+
+.. function:: bing_tile_coordinates(varchar) => array(bigint)
+
+    Given a QuadKey of a Bing tile, returns XY coordinates of that tile
```

Comment:
What is a "QuadKey"? It's not defined elsewhere in the documentation. Why is it a varchar? Is it something the user is expected to type by hand? Or should it be a type in Presto?

Reply:
@electrum ,

QuadKey is a unique tile identifier defined in https://msdn.microsoft.com/en-us/library/bb259689.aspx:

"To optimize the indexing and storage of tiles, the two-dimensional tile XY coordinates are combined into one-dimensional strings called quadtree keys, or “quadkeys” for short. Each quadkey uniquely identifies a single tile at a particular level of detail, and it can be used as an key in common database B-tree indexes. To convert tile coordinates into a quadkey, the bits of the Y and X coordinates are interleaved, and the result is interpreted as a base-4 number (with leading zeros maintained) and converted into a string."

I expect users to generate quadkeys by calling point_to_bing_tile and geometry_to_bing_tiles functions. These can then be stored and used in UI tools which display data per tile.

I can add a new BingTile type if that's beneficial. I imagine it would have 2 constructors: one takes quadkey, the other a tuple of (X, Y, zoomLevel). Let me know whether you think a BingTile type is necessary.

Reply:
Using a string seems like a rather inefficient internal representation. From my brief reading, its seems to only require at most 46 bits (23*2). So we can add a type for it that stores the value as a 64-bit number (extending `AbstractLongType`), probably treating the top/bottom 32-bits as the X/Y coordinates.

The type can render in the canonical string format when cast to varchar.